src/.github/workflows/{% if ci == 'GitHub' %}test.yml{% endif %}.jinja

```diff
@@ -29,6 +29,14 @@ set IMAGES = {
   }
 }
 -%}
+
+{%- if ci_extra_env -%}
+env:
+  {%- for key, value in ci_extra_env.items() %}
+  {{ key }}: {{ value|tojson }}
+  {%- endfor %}
+{%- endif %}
+
```

Comment:
Maybe it's better to have this in the `test` job only. Better be specific first and generalize later if needed.

Reply:
Moved to just the test job.